include/seqan3/core/type_traits/template_inspection.hpp

```diff
@@ -59,7 +59,7 @@ template <template <typename ...> typename source_template,
 //!\cond
     requires requires ()
     {
-        std::declval<target_template<source_arg_types...>>();
+        typename target_template<source_arg_types...>;
```

Comment:
why is the `typename` needed?

Reply:
Yes good question. If I see it correctly, then we just wanted to know before whether the type `target_template` substituted with the given `source_arg_types`  is actual a valid type definition. But in the concept defintion this can be expressed explicitly as a type requirement. Here is the excerpt of [cppreference](https://en.cppreference.com/w/cpp/language/constraints):

>A type requirement is the keyword typename followed by a type name, optionally qualified. The requirement is that the named type is valid: this can be used to verify that a certain named nested type exists, or that a class template specialization names a type, or that an alias template specialization names a type. A type requirement naming a class template specialization does not require the type to be complete.